packages/atclient/src/connection.c

```diff
@@ -788,6 +788,15 @@ int atclient_connection_read(atclient_connection *ctx, unsigned char **value, si
 exit: { return ret; }
 }
 
+void atclient_connection_set_read_timeout(atclient_connection *ctx, const uint32_t timeout_ms) {
+  if (ctx == NULL) {
+    atlogger_log(TAG, ATLOGGER_LOGGING_LEVEL_ERROR, "ctx is NULL\n");
+    return;
+  }
+
+  mbedtls_ssl_conf_read_timeout(&(ctx->ssl_config), timeout_ms);
+}
+
```

Comment:
Doesn't this method already exist? I recall editing this in the changes for #475